zql/zql.peg

```diff
@@ -218,25 +217,29 @@ procList
     }
 
 parallelChain
-  = _? ";" _? ch:procChain { RETURN(MAP("op": "SequentialProc", "procs": ch)) }
+  = __ ";" __ ch:procChain { RETURN(MAP("op": "SequentialProc", "procs": ch)) }
 
 proc
   = simpleProc
   / groupByProc
-  / "(" _? proc:procList _? ")" {
+  / "(" __ proc:procList __ ")" {
       RETURN(proc)
     }
 
 groupByKeys
-  = "by"i _ first:groupByKey rest:(__ "," __ cl:groupByKey { RETURN(cl) })* {
-      RETURN(PREPEND(first, rest))
-    }
-
+  = "by"i _ columns:FlexAssignments { RETURN(columns) }
 
-groupByKey
-  = ExpressionAssignment
-  / field:fieldExpr { RETURN(MAP("op": "ExpressionAssignment", "target": TEXT, "expression": field)) }
+// A FlexAssignment is like an Assignment but it can optionally omit the lhs,
+// in which case, the semantic pass will infer a name from the rhs, e.g., for
```

Comment:
```suggestion
// in which case the semantic pass will infer a name from the rhs. E.g., for
```

Reply:
Hmm I always put that comment in.  Need to break the habit.